examples/connected-components/component.c

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright (c) 2024, The Pallene Developers
+ * Pallene Tracer is licensed under the MIT license.
+ * Please refer to the LICENSE and AUTHORS files for details
+ * SPDX-License-Identifier: MIT
+ */
+
+#include <stdlib.h>
+
+#define PT_IMPLEMENTATION
+#include "ptracer.h"
+
+/* ---------------- PALLENE TRACER LUA INERFACE ---------------- */
+#define CON_LUA_FRAMEENTER(fnptr)                       \
+    PALLENE_TRACER_LUA_FRAMEENTER(L, fnstack, fnptr,    \
+        lua_upvalueindex(1), _frame)
+/* ---------------- PALLENE TRACER LUA INERFACE END ---------------- */
+
+/* ---------------- PALLENE TRACER C INTERFACE ---------------- */
+
+#define CON_C_FRAMEENTER()                              \
+    PALLENE_TRACER_GENERIC_C_FRAMEENTER(fnstack, _frame)
+
+#define CON_C_SETLINE()                                 \
+    PALLENE_TRACER_GENERIC_C_SETLINE(fnstack)
+
+#define CON_C_FRAMEEXIT()                               \
+    PALLENE_TRACER_FRAMEEXIT(fnstack)
```

Comment:
* These would look less scary if the defines fit in the same line, without `\`
* The PALLENE_TRACER_GENERIC names are way too long
* Even nicer would be if the user didn't have to define macros and could use ptracer macros directly.

Reply:
- Sure.
 - LOUD names are less confusing in my opinion. @srijan-paul also feel the same way I presume.
 - Users can, but it would be sorta hassle. We have talked about this earlier, it's okay if users define their own macros, it's not huge boilerplate. Besides, the usual way to go would be put this separately in a header file in module code base, so most C translations don't even have to look at these mess (see the multimod test). Writing some simple boilerplate shouldn't hurt, as you are writing frameenter and frameexit repeatedly in every function.

Reply:
My main issue with the names are

1. The long names blend together. In `PALLENE_TRACER_GENERIC_C_SETLINE`, the common PALLENE_TRACER_GENERIC prefix takes almost all the space and the relevant SETLINE is a small part of the end. 
2. We expect to call SETLINE very often, and such a long name makes it inconvenient to type by hand (without defining a helper macro)
3. One of the original ideas with the GENERIC macros was to have good defaults. We did that for __LINE__ numbers, but not for the fnstack. If we enforced a default for how to use the fnstack, or if we allowed the user to specify a "FNSTACK" macro, then there would be no need for CON_C_SETLINE or CON_C_FRAMEEXIT.

Reply:
They are not actually "issues" per-se, at-least to my eyes.

1. Alternative way to do this would be to reduce `PALLENE_TRACER_GENERIC` to something shorter, inevitably changing the `PALLENE_TRACER` part, which is prefix for almost everything in Pallene Tracer. Doing something like `PALLENE_TRACER_GCS` would be silly.
2. You are supposed to write some macros. Because no matter how hard we abstract, users have to define their own macros one way or the another for Pallene Tracer APIs because a lot of things are in users hands. Users hands won't fall off if they write some extra macros, not to mention they would be copying most of the part anyway.
3. Way too much generalized. What if users want to name it "stack" instead of "fnstack"? What if they want to make fnstack global? Or want to pass it as an upvalue? If we do account for all the cases, it would make macro matters even complicated. We did the generic part for setlines because it is only and only `__LINE__ + 1` for every users, which can be and should be generalized.

Reply:
1. We could use a shorter name like PTRACER_ or even just PT_
2. It's unfortunate that even the simplest examples need boilerplate macros
3. For fnstack, it might suffice to ask the user for a single macro that says the name of the fnstack variable. I'm also not against recommending a default name. The idea is to have good defaults for the common cases.